web/src/app/browser/src/contextManager.ts

```diff
@@ -306,23 +325,57 @@ export default class ContextManager extends ContextManagerBase<BrowserConfigurat
 
     let attachment = target.getElement()._kmwAttachment;
 
+    // Catches if the target is already in independent-mode, even if it's being cancelled
+    // during this call.
+    const wasPriorTarget = this.keyboardTarget == target;
+
     if(!attachment) {
       return;
     } else {
       // If directly set
       attachment.keyboard = kbdId || null;
       attachment.languageCode = langId || null;
 
-      if(this.keyboardTarget == target) {
-        this.activateKeyboard(attachment.keyboard, attachment.languageCode, true);
+      // If it has just entered independent-keyboard mode, we need the second check.
+      if(wasPriorTarget || this.keyboardTarget == target) {
```

Comment:
Perhaps? Just to clarify that the check is subject to side-effects that aren't apparent just from reading lines 330-340.

```suggestion
      if(wasPriorTarget || this.determineKeyboardTarget() == target) {
```